### PR TITLE
Tdes encryption fips disable

### PR DESCRIPTION
**Pull request by slontis**
Add FIPS indicator support for Triple-DES encryption.

This leaves 3DES with the FIPS query "FIPS=yes", which allows
Triple-DES to be used for Decryption by default.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: providers/implementations/ciphers/cipher_tdes_common.c
Comment (slontis) [inline providers/implementations/ciphers/cipher_tdes_common.c:114]: I would really like others opinions about where this check should happen..
Because of the stupid way the Init API works, this could get triggered multiple times when multiple init calls happen.

I could leave it to check in the update but that seems late.

Comment (slontis) [reply providers/implementations/ciphers/cipher_tdes_common.c:114]: e.g. If we wanted cmac_prov to have an indicator related to the cipher encrypt, cmac_prov.c calls CMAC_init() which does the encrypt_init internally so I would have to have another function if we wanted to allow CMAC_init to pass params to its init. Doing it during the update would be easier.

File: providers/implementations/macs/cmac_prov.c
Comment (slontis) [inline providers/implementations/macs/cmac_prov.c:214]: Is it ok to just do the simple strict check here?

File: providers/implementations/ciphers/cipher_tdes_common.c
Comment (nhorman) [reply providers/implementations/ciphers/cipher_tdes_common.c:114]: You could just call OSSL_FIPS_IND_ON_UNAPPROVED in ossl_tdes_einit, but I'm not sure how much that saves you

Comment (paulidale) [reply providers/implementations/ciphers/cipher_tdes_common.c:114]: > e.g. If we wanted cmac_prov to have an indicator related to the cipher encrypt

We definitely want CMAC to include the validity the the cipher used.  I don't think this is captured in the epic.

**Comment by slontis** [comment]
Fixed relevant code style issues

File: providers/implementations/macs/cmac_prov.c
Comment (paulidale) [reply providers/implementations/macs/cmac_prov.c:214]: I'd prefer a whitelist approach rather than blacklist.

File: doc/man1/openssl-fipsinstall.pod.in
Comment (paulidale) [inline doc/man1/openssl-fipsinstall.pod.in:196]: Ditch the parenthesis.  A full sentence should never be parenthesised by itself.

File: providers/fips/fipsprov.c
Comment (paulidale) [inline providers/fips/fipsprov.c:366]: I think it would be possible to drop the `UNAPPROVED_ALG` macro since this is it's only use.

**Review by paulidale — CHANGES_REQUESTED** [review]

File: providers/implementations/macs/cmac_prov.c
Comment (slontis) [reply providers/implementations/macs/cmac_prov.c:214]: Will change - its a bit slower.

File: providers/fips/fipsprov.c
Comment (slontis) [reply providers/fips/fipsprov.c:366]: Other macro removed also, they can be re-added if required later.

**Review by paulidale — APPROVED** [review]

**Review by nhorman — APPROVED** [review]

**Comment by slontis** [comment]
rebased

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
rebased - I will try to merge this if the tests pass.

**Pull request closed without merging**
